### PR TITLE
[integration] Hackaton fix: pilotsubmission discrepancies between Accounting and Monitoring

### PR DESCRIPTION
**Pull request by rupozzi**
Fixing the discrepancies between the plots in accounting and monitoring for PilotSubmissions

BEGINRELEASENOTES

*Accounting/Monitoring
FIX: Fixed the plotter for pilot submission accounting and other small fixes

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py [hunk 1]
```diff
@@ -164,6 +165,18 @@ def commit(self):
             self.__documentLock.release()
         return S_OK(recordSent)
 
+    def delayedCommit(self):
```
Comment (fstagni) [inline src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py:168]: is this used?

File: src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py [hunk 2]
```diff
@@ -164,6 +165,18 @@ def commit(self):
...
+        """
+        If needed start a timer that will run the commit later
+        allowing to send more registers at once (reduces overheads).
+        """
+
+        if not self.__commitTimer.is_alive():
```
Comment (chrisburr) [inline src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py:174]: Should this do something to make sure `self.commit` is always called, even if the python process shuts down?

Comment (rupozzi) [reply src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py:174]: I mainly mirrored what is done for Accounting... What would you suggest?

File: src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py [hunk 3]
```diff
@@ -49,6 +49,7 @@ def __init__(self, monitoringType="", failoverQueueName="dirac.monitoring"):
         self.__monitoringType = monitoringType
         self.__failoverQueueName = failoverQueueName
         self.__defaultMQProducer = None
+        self.__commitTimer = threading.Timer(5, self.commit)
```
Comment (fstagni) [inline src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py:52]: This does not seem used.

Comment (rupozzi) [reply src/DIRAC/MonitoringSystem/Client/MonitoringReporter.py:52]: You are right. Forgot it there when I wanted to implement the delayedCommit() method,

**Review by fstagni — APPROVED** [review]

**Pull request merged**